### PR TITLE
Docs setup part modularisation

### PR DESCRIPTION
**Pull request by MuskanPaliwal**
- Created different components for each step such as `triggerContent`, `buildContent`, `hostingSelector`, `providerSelector`, `installationSelector` and `CodeWithCopyButton` components. 

## Summary by CodeRabbit

- **New Features**
  - Introduced new components for an enhanced setup experience, including specific selectors for cloud providers, hosting options, and installation types.

- **Refactor**
  - Improved the documentation structure for better user guidance through the setup process.

- **Documentation**
  - Updated the Docs section with clearer build instructions and trigger content.

### CODE REVIEW

File: pages/docs/index.tsx
Comment (avikalpg) [inline pages/docs/index.tsx:10]: If this icon is not used, please remove it.

File: components/Setup/BuildInsruction.tsx
Comment (avikalpg) [inline components/Setup/BuildInsruction.tsx:1]: Can you please rename the folder is smallcase/camelCase `setup` instead of `Setup`?

I am trying to adhere to a naming convention where only Classes and Objects are written in PascalCase and everything else is written in camelCase (this is also the industry standard in JS/TS).

**Review by avikalpg — CHANGES_REQUESTED** [review]
The pull request looks good to me. I am assuming that the functionality is unchanged and that it is tested for it.

I only have 2 minor suggestions. Once you accommodate them, I will approve the pull request.

File: components/Setup/BuildInsruction.tsx
Comment (MuskanPaliwal) [reply components/Setup/BuildInsruction.tsx:1]: Sure I will do it right now

**Comment by MuskanPaliwal** [comment]
> The pull request looks good to me. I am assuming that the functionality is unchanged and that it is tested for it.
> 
> I only have 2 minor suggestions. Once you accommodate them, I will approve the pull request.

Yess, I tested the whole functionality, it's working. Just a few things related to env vars to be fixed in the main branch. I will raise another PR for that

**Review by tapishr — COMMENTED** [review]
Looking good to me, can be merged after avikalp's requested changes

**Review by tapishr — APPROVED** [review]

**Review by avikalpg — APPROVED** [review]

**Pull request merged**
